pkg/helper/envvarutils.go

```diff
@@ -107,3 +107,18 @@ func EnsureEnvVar(desired v1.EnvVar, existingEnvVars *[]v1.EnvVar) bool {
 
 	return update
 }
+
+// RemoveDuplicateEnvVars removes duplicate env vars by name from a slice
+func RemoveDuplicateEnvVars(envVars []v1.EnvVar) []v1.EnvVar {
+	set := map[string]v1.EnvVar{}
+	var result []v1.EnvVar
+
+	for idx := range envVars {
+		if _, ok := set[envVars[idx].Name]; !ok {
+			set[envVars[idx].Name] = envVars[idx]
+			result = append(result, envVars[idx])
+		}
+	}
+
+	return result
```

Comment:
Help me understand where in this function the env vars are actually removed.

Reply:
The map has the property that the map keys are unique. The method will iterate over all the env vars and try to create a map of env var key to the same env var.
```
{env var key A} -> {env var A}
{env var key B} -> {env var B}
{env var key C} -> {env var C}
```
On each iteration of the loop, the map can tell if the key has been inserted before or not (line 17) (in O(1) time). When there are duplicated env vars, the loop will try to insert the same key multiple times in the map. Only the first time will succeed. After that, when a duplicated key is found, the conditional (line 17) will evaluate to false and the duplicated env var will not be added to the list of env vars that is returned by the method.